### PR TITLE
[futures] add a few blanket impls to std

### PR DESCRIPTION
**Pull request by tinaun**
these were defined in the futures crate, but with the core definitions moving to std these would need to move too.

### CODE REVIEW

File: src/liballoc/boxed.rs [hunk 1]
```diff
@@ -914,6 +914,45 @@ impl<T: ?Sized + Unsize<U>, U: ?Sized> CoerceUnsized<PinBox<U>> for PinBox<T> {}
...
+impl<'a, F: ?Sized + Future> Future for PinBox<F> {
+    type Output = F::Output;
+
+    fn poll(mut self: PinMut<Self>, cx: &mut Context) -> Poll<Self::Output> {
+        self.as_pin_mut().poll(cx)
+    }
+}
+
+#[unstable(feature = "futures_api", issue = "50547")]
+unsafe impl<F: Future<Output = ()> + Send + 'static> UnsafePoll for Box<F> {
```
Comment (cramertj) [inline src/liballoc/boxed.rs]: I don't think this impl is needed anymore-- I think people should prefer to use `PInBox`, as it offers the same functionality in a safer way.

**Comment by cramertj** [comment]
Thanks for adding these! Are you planning to add the missing methods on `Poll` in a separate PR?

**Comment by cramertj** [comment]
r? cramertj

**Comment by tinaun** [comment]
right now they exist as an extension trait in futures-core, but if there’s nothing controversial about them i can move them in

**Comment by cramertj** [comment]
@tinaun I think it's fine to move them in-- they don't seem particularly controversial to me (thinking of `map`, `is_ready`, `is_pending`, `map_ok`, and `map_err`).

**Pull request merged**